resources/todomvc/architecture-examples/backbone/src/sync/backbone.sync.js

```diff
@@ -31,6 +31,8 @@ Backbone.sync = (function (method, model, options) {
             size += 1;
 
             // if a success callback was provided, execute it.
+
+            // prettier-ignore
```

Comment:
I'm happy that this change allows us to run prettier again but it's unfortunate that single line conditions require header text to work (and likely hard to remember when writing patches). What are our options here? I could imagine some alternatives

1) Some tooling changes allow us to reformat single line conditions after prettier runs (maybe an eslint plugin etc)
2) We change the formatting guidelines to put brackets around single line conditions
3) We stop using prettier on js files and only use eslint

I think Thorsten has looked into (1) a bit already and this would be my least favorite option as it sounds like it will take engineering time to solve (as opposed to a configuration change).

(2) would be the simplest technically and is my personal preference - it's a style decision we could decide on and then just re-run prettier as is. I know we've had discussions in the past about changes from the webkit guidelines around bracket style with functions, but we didn't extend that to this single-line conditional rule. I would have brought it up at the time when but didn't realize there was still a difference with what tooling was producing. Perhaps @rniwa is open to making this change?

What are the implications of (3)? I see prettier fixed some whitespace in main-container.jsx but given that the majority of this patch is undoing these conditions maybe it isn't buying us a lot over what's currently getting written?

Reply:
> I'm happy that this change allows us to run prettier again but it's unfortunate that single line conditions require header text to work (and likely hard to remember when writing patches). What are our options here? I could imagine some alternatives
> 
> 1. Some tooling changes allow us to reformat single line conditions after prettier runs (maybe an eslint plugin etc)
> 2. We change the formatting guidelines to put brackets around single line conditions
> 3. We stop using prettier on js files and only use eslint
> 
> I think Thorsten has looked into (1) a bit already and this would be my least favorite option as it sounds like it will take engineering time to solve (as opposed to a configuration change).
> 
> (2) would be the simplest technically and is my personal preference - it's a style decision we could decide on and then just re-run prettier as is. I know we've had discussions in the past about changes from the webkit guidelines around bracket style with functions, but we didn't extend that to this single-line conditional rule. I would have brought it up at the time when but didn't realize there was still a difference with what tooling was producing. Perhaps @rniwa is open to making this change?

I'd prefer anything but (2). I don't want our style to keep deviating from WebKit's coding style guideline since that'll make contributing to Speedometer harder for WebKit contributors.

Reply:
@flashdesignory actually I think the eslint fixing is working fine as-is with the "curly" and "nonblock-statement-body-position" rules. The problem seems to be that it's only running on mjs files (from the package.json `eslint **/*.mjs --fix`). We can see all the files that are requiring the prettier ignore in this script are .js or .jsx. Perhaps we can fix this by changing the task to include more file extensions?

Reply:
ha - will do 😄

Reply:
@bgrins @rniwa - adjusted the settings a bit and now opting in js, mjs and jsx files.

Reply:
nice cleanup! the lint job is failing in CI https://github.com/WebKit/Speedometer/actions/runs/4493309669/jobs/7904354382?pr=113 but otherwise LGTM